access/validator.go

```diff
@@ -192,6 +204,33 @@ func (v *TransactionValidator) checkCanBeParsed(tx *flow.TransactionBody) error
 	return nil
 }
 
+func (v *TransactionValidator) checkAddresses(tx *flow.TransactionBody) error {
+
+	for _, address := range append(tx.Authorizers, tx.Payer) {
```

Comment:
Isn't the payer's responsibility to check the authorizers are valid? I mean shouldn't the payer still pay for a transaction that has invalid authorizers?

Reply:
It is the payer's responsibility to some extent, but we still also want to not include invalid transactions as well. There is a whole class of spam where payers sign an invalid transaction for which they don't have enough funds to pay. Plus, be rejecting these transactions here we reduce overall load on the rest of the network. In general the collection nodes should filter out as much invalid inputs as they can.

Reply:
For payers with enough balance, it makes sense to make them pay as a countermeasure of spamming collectors with invalid transactions. I agree we should filter the payers without balance early in the pipeline.
Till we implement collector's access to the execution state, I think your approach is right as I guess a spamming payer is more likely to be an account with low/no balance.

Reply:
> For payers with enough balance, it makes sense to make them pay as a countermeasure of spamming collectors with invalid transactions

Good point 👍